dds/idl/marshal_generator.cpp

```diff
@@ -3518,14 +3518,14 @@ bool marshal_generator::gen_union(AST_Union* node, UTL_ScopedName* name,
 
     // Add a reference to the idl file, if the descriminator is user defined.
     AST_Type* disc_type = resolveActualType(discriminator);
-    const Classification disc_cls = classify(disc_type);
+    const Classification disc_cls2 = classify(disc_type);
     if (!disc_type->in_main_file() && disc_type->node_type() != AST_Decl::NT_pre_defined) {
       be_global->add_referenced(disc_type->file_name().c_str());
     }
 
     // Determine the default enum value
     ACE_CDR::ULong default_enum_val = 0;
-    if (disc_cls & CL_ENUM) {
+    if (disc_cls2 & CL_ENUM) {
```

Comment:
```suggestion
    if (disc_cls & CL_ENUM) {
```

I don't think this needs to change